pointillism-port.patch

```diff
@@ -0,0 +1,129 @@
+diff --git a/activity.py b/activity.py
+index d91ea44..c4edf8b 100755
+--- a/activity.py
++++ b/activity.py
+@@ -1,4 +1,4 @@
+-#! /usr/bin/env python
++#! /usr/bin/env python3
+ # -*- coding: utf-8 -*-
+ #
+ # Pointillism
+@@ -24,13 +24,20 @@
+ 
+ import puntillism
+ from gettext import gettext as _
+-from sugar3.datastore import datastore
++# from sugar3.datastore import datastore
+ from sugar3.graphics.objectchooser import ObjectChooser
+ from sugar3.graphics.toolbutton import ToolButton
+ from sugar3.activity.widgets import StopButton
+ from sugar3.activity.widgets import ActivityToolbarButton
+ from sugar3.graphics.toolbarbox import ToolbarBox
+-from sugar3.activity import activity
++# from sugar3.activity import activity
++import shutil
++import sys
++from sugarapp.widgets import SugarCompatibleActivity
++from sugarapp.widgets import DesktopSaveChooser
++from sugarapp.widgets import DesktopOpenChooser
++from sugar3.graphics.style import GRID_CELL_SIZE
++from sugarapp.helpers import PrimaryMonitor
+ from sugar3 import mime
+ import sugargame.canvas
+ import sugargame
+@@ -43,10 +50,10 @@ gi.require_version('Gdk', '3.0')
+ gi.require_version('Gtk', '3.0')
+ 
+ 
+-class Activity(activity.Activity):
++class Activity(SugarCompatibleActivity):
+ 
+     def __init__(self, handle):
+-        activity.Activity.__init__(self, handle)
++        SugarCompatibleActivity.__init__(self, handle)
+ 
+         self.max_participants = 1
+         self.file_path_temp = None
+@@ -56,13 +63,16 @@ class Activity(activity.Activity):
+         self.actividad = puntillism.Puntillism(self)
+ 
+         self.build_toolbar()
+-
++        
++        w = PrimaryMonitor.width()
++        h = PrimaryMonitor.height() - 1 * GRID_CELL_SIZE
++ 
+         self._pygamecanvas = \
+             sugargame.canvas.PygameCanvas(
+                 self,
+                 main=self.actividad.run,
+                 modules=[pygame.display])
+-
++        self._pygamecanvas.set_size_request(w, h)
+         self.set_canvas(self._pygamecanvas)
+         self._pygamecanvas.grab_focus()
+ 
+@@ -155,19 +165,18 @@ class Activity(activity.Activity):
+     def _savebutton_cb(self, button):
+         pygame.event.post(pygame.event.Event(
+             pygame.USEREVENT, action='savebutton'))
+-
++    
++    def can_close(self):
++        self.actividad.running = False
++        self.load_image_loop = False
++        pygame.quit()
```

Comment:
Any particular reason for leaving "pygame.quit()" here?

Reply:
pygame.quit() is safe ending of the game. It is used by almost all games, it closes the display